addon/@ember/test-waiters/build-waiter.ts

```diff
@@ -3,13 +3,33 @@ import { Primitive, TestWaiter, TestWaiterDebugInfo, WaiterName } from './';
 import { DEBUG } from '@glimmer/env';
 import { warn } from '@ember/debug';
 import Token from './token';
-import { register } from './waiter-manager';
+import { register, getPrivateData } from './waiter-manager';
 
 const WAITER_NAME_PATTERN = /^[^:]*:?.*/;
-let WAITER_NAMES = DEBUG ? new Set() : undefined;
```

Comment:
the core of this PR is moving `WAITER_NAMES` from module-state to state that lives on `globalThis` so that it can be shared

Reply:
It seems like we don't really need this at all. It appears that the purpose of this is just to keep track of what names has been already been used and issue a warning. But in all cases, immediately the waiter is registered on the `Map` with that string anyway, so it seems like we could have just checked the map for an existing entry in `register`?

Reply:
> Is this theoretical/hypothetical, or were you actually observing the log spew in tests or something? 


This is preparing for switching the library to a v2 addon (See ["The Plan"](https://github.com/emberjs/ember-test-waiters/issues/458)), where we'll have no more highlander code.

Today, with the v1 addon, the WAITER_NAMES stuff is a non-issue because there is only one copy of `@ember/test-waiters` -- that will likely not be the case with native v2 addons.

As for hypothetical -- we're saying that "all waiters, across all parts of the module graph, must have unique names".

Reply:
Ok I was hedging a little bit there and maybe that phrasing confused you. I am saying I didn’t see any evidence that this code is specifically trying to avoid log spew even going back to the original commit. So I am asking where did that concern come from.

Can you read my [comment](https://github.com/emberjs/ember-test-waiters/pull/461#discussion_r1413734731) again? I was specifically saying/clarify “it doesn’t seem like we would have to worry about log spew here based on the usage patterns, is there a specific reason you are concerned?”

I was quite specific about why I don’t think that is an issue here and why I didn’t think this was a specific concern when the code was originally added.

I could definitely be wrong but I am asking where did the concern of log spew came from - was it from deductions from analyzing the code, or from discovering the problem when running the tests, etc.